packages/netstring/test/test-netstring.js

```diff
@@ -0,0 +1,114 @@
+// @ts-check
+
+import test from 'ava';
+import { netstringReader } from '../src/reader.js';
+import { netstringWriter } from '../src/writer.js';
+import { pipe } from './stream.js';
+
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
+
+async function read(source) {
+  const array = [];
+  for await (const chunk of source) {
+    // Capture current state, allocating a copy.
+    array.push(chunk.slice());
+  }
+  return array;
+}
+
+test('read short messages', async t => {
+  const r = netstringReader([encoder.encode('0:,1:A,')], '<unknown>', 1);
+  const array = await read(r);
+  t.deepEqual(
+    ['', 'A'],
+    array.map(chunk => decoder.decode(chunk)),
+  );
+});
+
+test('read messages divided over chunk boundaries', async t => {
+  const r = netstringReader(
+    [encoder.encode('5:hel'), encoder.encode('lo,')],
+    '<unknown>',
+    1,
+  );
+  const array = await read(r);
+  t.deepEqual(
+    ['hello'],
+    array.map(chunk => decoder.decode(chunk)),
+  );
+});
+
+function delay(ms) {
+  return new Promise(resolve => {
+    setTimeout(resolve, ms);
```

Comment:
`setTimeout` is a powerful global; how does this pass `lint` without `/* global setTimeout */`?

Reply:
https://github.com/endojs/endo/issues/696